### PR TITLE
emmet-language-server: init at v2.2.0

### PR DESCRIPTION
**Pull request by stnley**
## Description of changes
Adding package https://github.com/olrtg/emmet-language-server

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/language-servers/emmet-language-server/default.nix
Comment (t4ccer) [inline pkgs/development/tools/language-servers/emmet-language-server/default.nix:9]:
```suggestion
  version = "2.2.0";
```
Version should be just a number, no `v` prefix, see https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

File: pkgs/development/tools/language-servers/emmet-language-server/default.nix
Comment (t4ccer) [inline pkgs/development/tools/language-servers/emmet-language-server/default.nix:13]:
```suggestion
    repo = "emmet-language-server";
```
Usually the change of `pname` shouldn't influence the change in source. More info https://github.com/nix-community/nixpkgs-lint/issues/21

File: pkgs/development/tools/language-servers/emmet-language-server/default.nix
Comment (t4ccer) [inline pkgs/development/tools/language-servers/emmet-language-server/default.nix:14]:
```suggestion
    rev = "v${version}";
```
Once you get rid of `v` from version, you must add it here to match the tag

File: pkgs/development/tools/language-servers/emmet-language-server/default.nix
Comment (t4ccer) [inline pkgs/development/tools/language-servers/emmet-language-server/default.nix:30]: You should add yourself to maintainers

File: pkgs/development/tools/language-servers/emmet-language-server/default.nix
Comment (t4ccer) [inline pkgs/development/tools/language-servers/emmet-language-server/default.nix:26]:
```suggestion
    description = "A language server for emmet.io";
```
Remove the trailing space

File: pkgs/development/tools/language-servers/emmet-language-server/default.nix
Comment (t4ccer) [inline pkgs/development/tools/language-servers/emmet-language-server/default.nix:4]:
```suggestion
```
You don't seem to be using it anywhere

**Review by t4ccer — CHANGES_REQUESTED** [review]

**Comment by stnley** [comment]
@t4ccer Thanks for the review. Should I squash those changes into 1 commit (with init message) then force push?

**Comment by t4ccer** [comment]
Yes please. You can see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions for details, but all changes for one "unit" should be in a single commit

**Comment by stnley** [comment]
Resolved notes from review and pushed new commits.

**Comment by stnley** [comment]
@t4ccer Just checking to see if the new changes look good

**Comment by t4ccer** [comment]
Oh one more thing. The commit message should say `emmet-language-server: init at 2.2.0` (without `v`)

**Comment by stnley** [comment]
@t4ccer Right! Fixed and pushed once again

**Review by t4ccer — APPROVED** [review]
LGTM! You'd need to wait for someone with write access to look again and merge though

**Review by emilytrau — APPROVED** [review]
Apologies for the delay! Thank you for your contribution

**Pull request merged**
